src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -54,7 +55,7 @@ public static async Task<FunctionMetadataResponse> ToFunctionMetadataResponse(th
                 response.ConfigHref = VirtualFileSystem.FilePathToVfsUri(functionMetadataFilePath, baseUrl, hostOptions);
             }
 
-            if (!string.IsNullOrEmpty(hostOptions.TestDataPath))
+            if (!string.IsNullOrEmpty(hostOptions.TestDataPath) && !excludeTestData)
```

Comment:
Check the flag before checking the string here

Reply:
Order swapped.